File: tests/fixtures/dogfood/scala_real_repo/app/controllers/UserController.scala
// Dogfood real-repo fixture for eshu-hq/eshu#5399.
//
// Play controller-action shape informed by public patterns in
// `playframework/playframework` at pinned SHA
// bcdc682de2250bbd0f2788bc5acc06f6d66ad5a7 (provenance metadata only, never
// fetched). Synthetic content, hand-authored for this fixture.

package controllers

import models.User
import services.UserService

class UserController(userService: UserService) {

  def show(id: String): User = {
    userService.getUser(id)
  }

  def create(name: String, email: String): User = {
    userService.createUser(name, email)
  }

  def healthCheck(): String = {
    "ok"
  }
}

object UserControllerApp extends App {
  val repository = new services.InMemoryUserRepository()
  val service = new UserService(repository)
  val controller = new UserController(service)
  controller.create("dogfood", "dogfood@example.invalid")
  println(controller.healthCheck())
}

File: tests/fixtures/dogfood/java_real_repo/src/main/java/com/example/dogfood/UserController.java
// Dogfood real-repo fixture for eshu-hq/eshu#5399.
//
// Spring MVC @RestController route shape informed by public patterns in
// Spring Boot-style services. Synthetic content, hand-authored for this
// fixture.

package com.example.dogfood;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {
    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/users/{id}")
    public User show(@PathVariable String id) {
        return userService.getOrThrow(id);
    }

    @PostMapping("/users")
    public User create(@RequestParam String name, @RequestParam String email) {
        return userService.create(name, email);
    }

    @GetMapping("/health")
    public String healthCheck() {
        return "ok";
    }
}
